File: November Long 2022 Division 4 (Rated)/e.py
modulo = 10**9 + 7
MAX = 10**6
facts = [0] * (MAX + 1)
facts[0] = 1
facts[1] = 1
s = 1
for i in range(2, MAX + 1):
    s = ((s % modulo) * (i % modulo)) % modulo
    facts[i] = s


for _ in range(int(input())):
	n = input()
	a = list(map(int, input().split()))
	asum = 0
	for i in a:
		asum += facts[i]

	print(asum % modulo)

File: November Long 2022 Division 4 (Rated)/e.cpp
#include <iostream>
using namespace std;
#define ll long long
#define mod 1000000007
#define MAX 1000001
ll facts[MAX + 1];

void precompute()
{
    facts[0] = 1;
    facts[1] = 1;
    ll s = 1;
    for (int i = 2; i <= MAX; i++)
    {
        s = ((s % mod) * (i % mod)) % mod;
        facts[i] = s;
    }
}

int main()
{
    precompute();
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        ll asum = 0;
        for (int i = 0; i < n; i++)
        {
            int x;
            cin >> x;
            asum += facts[x];
        }
        cout << asum % mod << endl;
    }
    return 0;
}